src/util/nidmap.c

```diff
@@ -1053,32 +1053,37 @@ int prrte_util_generate_ppn(prrte_job_t *jdata,
     prrte_proc_t *proc;
     size_t sz;
     prrte_buffer_t bucket;
+    prrte_app_context_t *app;
 
     PRRTE_CONSTRUCT(&bucket, prrte_buffer_t);
 
     for (i=0; i < jdata->num_apps; i++) {
         /* for each app_context */
-        for (j=0; j < jdata->map->nodes->size; j++) {
-            if (NULL == (nptr = (prrte_node_t*)prrte_pointer_array_get_item(jdata->map->nodes, j))) {
-                continue;
-            }
-            if (NULL == nptr->daemon) {
-                continue;
-            }
-            ppn = 0;
-            for (k=0; k < nptr->procs->size; k++) {
-                if (NULL != (proc = (prrte_proc_t*)prrte_pointer_array_get_item(nptr->procs, k))) {
-                    if (proc->name.jobid == jdata->jobid) {
-                        ++ppn;
-                    }
+        if (NULL != (app = (prrte_app_context_t*)prrte_pointer_array_get_item(jdata->apps, i))) {
+            for (j=0; j < jdata->map->num_nodes; j++) {
+                if (NULL == (nptr = (prrte_node_t*)prrte_pointer_array_get_item(jdata->map->nodes, j))) {
+                    continue;
+                     
                 }
-            }
-            if (0 < ppn) {
-                if (PRRTE_SUCCESS != (rc = prrte_dss.pack(&bucket, &nptr->index, 1, PRRTE_STD_CNTR))) {
-                    goto cleanup;
+                if (NULL == nptr->daemon) {
+                    continue;
                 }
-                if (PRRTE_SUCCESS != (rc = prrte_dss.pack(&bucket, &ppn, 1, PRRTE_UINT16))) {
-                    goto cleanup;
+                ppn = 0;
+                
+                for (k=0; k < (int) nptr->procs->size; k++) {
+                    if (NULL != (proc = (prrte_proc_t*)prrte_pointer_array_get_item(nptr->procs, k))) {
+                        if (proc->app_idx == app->idx) {
```

Comment:
@rhc54 it's this comparison that was giving me trouble.

I'd be getting, say, a count of 11 instead of 10 for something like `prun --np 10 hostname`, and I suspect that is because the daemon gets inserted into nptr->procs as well (also at idx 0). That's why I changed it to UINT32_MAX instead of 0, so that won't get caught here.

Reply:
Check the proc jobid instead of just the app idx - you need to do this anyway as PRRTE is persistent and may have many jobs running on it

Reply:
I think that will break multi-app jobs. I'll give it a shot - but I suspect it won't work.

Reply:
Checking the app idx and jobid seems to work well.